File: trackr/src/main/java/usecases/EventOutputData.java
package usecases;

import entities.Event;

import java.time.LocalDate;
import java.util.Set;

public class EventOutputData {
    private final String firstName;
    private final String lastName;
    private final String personDescription;
    private final Set<String> tags;
    private final LocalDate date;
    private final LocalDate remindDeadline;

    public EventOutputData(Event event) {
        this.firstName = event.getPerson().getFirstName();
        this.lastName = event.getPerson().getLastName();
        this.personDescription = event.getPerson().getDescription();
        this.tags = event.getPerson().getTags();
        this.date = event.getDate();
        this.remindDeadline = event.getReminderDeadline();
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getPersonDescription() {
        return personDescription;
    }

    public Set<String> getPersonTags() {
        return tags;
    }

    public LocalDate getDate() {
        return date;
    }

    public LocalDate getRemindDeadline() {
        return remindDeadline;
    }
}

File: trackr-app/app/src/main/java/com/trackr/trackr_app/usecases/EventOutputData.kt
package usecases

import com.trackr.trackr_app.entities.Anniversary
import com.trackr.trackr_app.entities.Event
import java.time.LocalDate

/**
 * A dataclass representing the Data that Presenter will retrieve from EventManager
 */
class EventOutputData(event: Event?) {
    private var eventType: EventTypes? = null
    var firstName: String? = null
    var lastName: String? = null
    var personDescription: String? = null
    var personTags: Set<String?>? = null
    var date: LocalDate? = null
    var remindDeadline: LocalDate? = null

    enum class EventTypes {
        ANNIVERSARY, BIRTHDAY
    }

    fun getEventType(): String {
        return eventType.toString()
    }

    init {
        if (event != null) {
            eventType = if (event is Anniversary) EventTypes.ANNIVERSARY else EventTypes.BIRTHDAY
            firstName = event.person.firstName
            lastName = event.person.lastName
            personDescription = event.person.description
            personTags = event.person.getTags()
            date = event.date
            remindDeadline = event.reminderDeadline
        }
    }
}